src/Sarif.Multitool/Rules/SARIF1008.PhysicalLocationPropertiesMustBeConsistent.cs

```diff
@@ -7,25 +7,50 @@ namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+        ///
+        /// 'region' allows both users and tools to distinguish similar results within the same artifact.If a SARIF viewer has access to the artifact, it can display it, and highlight the location identified by the analysis tool.If the region has a 'snippet' property, then even if the viewer doesn't have access to the artifact (which might be the case for a web-based viewer), it can still display the faulty code.
+        /// 
+        /// 'contextRegion' provides users with a broader view of the result location. Typically, it consists of a range starting a few lines before 'region' and ending a few lines after. Again, if a SARIF viewer has access to the artifact, it can display it, and highlight the context region (perhaps in a lighter shade than the region itself). This isn't terribly useful since the user can already see the whole file, with the 'region' already highlighted. But if 'contextRegion' has a 'snippet' property, then even a viewer without access to the artifact can display a few lines of code surrounding the actual result, which is helpful to users.
+        /// 
+        /// If the SARIF validator reports that 'contextRegion' is present but 'region' is absent, then it's possible that the tool should have populated 'region' rather than 'contextRegion', or that it simply neglected to populate 'region'. If the validator reports that 'contextRegion' is not a proper superset of 'region', then it's possible that the tool reversed 'region' and 'contextRegion'. If 'region' and 'contextRegion' are identical, the tool should simply omit
+        /// </summary>
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString { Text = RuleResources.SARIF1008_PhysicalLocationPropertiesMustBeConsistent_FullDescription_Text };
+
         protected override IEnumerable<string> MessageResourceNames => new string[]
         {
-            nameof(RuleResources.SARIF1008_PhysicalLocationPropertiesMustBeConsistent_Error_ContextRegionRequiresRegion_Text)
+            nameof(RuleResources.SARIF1008_PhysicalLocationPropertiesMustBeConsistent_Error_ContextRegionRequiresRegion_Text),
+            nameof(RuleResources.SARIF1008_PhysicalLocationPropertiesMustBeConsistent_Error_ContextRegionMustBeProperSupersetOfRegion_Text)
         };
 
+        public override FailureLevel DefaultLevel => FailureLevel.Error;
+
         protected override void Analyze(PhysicalLocation physicalLocation, string physicalLocationPointer)
         {
-            if (physicalLocation.ContextRegion != null && physicalLocation.Region == null)
+            if (physicalLocation.ContextRegion == null)
+            {
+                return;
+            }
+
+            // ContextRegionRequiresRegion: If 'contextRegion' is present, then 'region' must also be present.
```

Comment:
>If 'contextRegion' is present, then 'region' must also be present. [](start = 44, length = 66)

what is the source of this text? i guessed from Eddy's PR that it was written by himself (not picked from message strings)

Reply:
Anything you want, and what you wrote is good.

---
In reply to: [445040282](https://github.com/microsoft/sarif-sdk/pull/1925#discussion_r445040282) [](ancestors = 445040282)